android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1703,40 +1707,34 @@ public void onLocationError(int errorCode)
   @Override
   public void onLocationNotFound()
   {
-    showLocationNotFoundDialog();
+    if (mLocationErrorDialogAnnoying || (mLocationErrorDialog != null && mLocationErrorDialog.isShowing()))
+      return;
+
+    final String message = String.format("%s\n\n%s", getString(R.string.current_location_unknown_message),
+        getString(R.string.current_location_unknown_title));
+    mLocationErrorDialog = new AlertDialog.Builder(this)
+        .setMessage(message)
+        .setOnDismissListener(dialog -> mLocationErrorDialog = null)
+        .setNegativeButton(R.string.current_location_unknown_stop_button, (dialog, which) ->
+        {
+          LocationHelper.INSTANCE.setStopLocationUpdateByUser(true);
+          LocationHelper.INSTANCE.stop();
+        })
+        .setPositiveButton(R.string.current_location_unknown_continue_button, (dialog, which) ->
+        {
+          if (!LocationHelper.INSTANCE.isActive())
+            LocationHelper.INSTANCE.start();
+          LocationHelper.INSTANCE.switchToNextMode();
+          mLocationErrorDialogAnnoying = true;
```

Comment:
Why mLocationErrorDialogAnnoying is set here? With it enabled, user will see the dialog only once, without an option to stop location detection later, right? Would it be possible to _stop_ location detection by pressing a rotating radar icon in that case?

Reply:
After pressing "Continue" button this dialog will not be shown again in the current session. This state is not persisted. After restarting the app, it can appear again. Otherwise, it will be shown every 30 seconds until the location is found. In addition, please note that this flag is shared with another "location error" dialog. 

Stopping location by pressing radar icon is possible.

This patch has one minor glitch is that radar icon is not rotating when the dialog is on the screen. After closing the dialog by pressing "Continue" it will continue to rotate. This is UI problem, the location is still enabled in the background when dialog is on the screen. I didn't found how to fix this issue quickly.

Reply:
The glitch is a minor issue. The important point is that it would be still possible to stop location. The bad thing is that the behavior will change:
1. User presses the button and searches for location.
2. User sees a dialog "Stop" or "Continue".
3. User presses "Continue".
4. User presses the button and stops the location search.
5. User presses the button and searches for location.
6. [SURPRISE] There is no dialog anymore here. Location is searched for indefinitely, right?